src/Language/DifferentialDatalog/Syntax.hs

```diff
@@ -230,6 +232,22 @@ instance WithPos ArgType where
 instance PP ArgType where
     pp ArgType{..} = (if atypeMut then "mut" else empty) <+> pp atypeType
 
+data IdentifierWithPos = IdentifierWithPos { idPos :: Pos, idName :: String }
+makeIdentifierWithPos str = IdentifierWithPos nopos str
```

Comment:
I don't think it's a very good function name. It's more like `makeIdentifierWithOUTPos` :)

For better or worse, I tend not to use this type of helpers throughout the compiler. So perhaps just write `IdentifierWithPos nopos name`. I won't insist though.